templates/common/_base/units/machine-config-daemon-firstboot.service

```diff
@@ -0,0 +1,19 @@
+name: "machine-config-daemon-firstboot.service"
+enabled: true
+contents: |
+  [Unit]
+  Description=Machine Config Daemon Firstboot
+  # This effectively disables this unit unit we patch the MCS, so
+  # we can land this code and then get it built into the host.
+  ConditionPathExists=!/etc/pivot/image-pullspec
+  ConditionPathExists=/etc/ignition-machine-config-encapsulated.json
+  After=ignition-firstboot-complete.service
```

Comment:
While I was testing this locally, noticed that `machine-config-daemon-firstboot.service` runs when a newly created cluster reboots after applying ignition configs and updates, which leads to another reboot. It's because /etc/pivot/image-pullspec gets deleted once cluster is updated to latest machine-os-content during first boot.  Should we also add something like `BindsTo=ignition-firstboot-complete.service` here to avoid running it in next reboot or it's an acceptable behavior?

Reply:
Hmm...we should be deleting `/etc/pivot/image-pullspec` during firstboot.

Reply:
yes, we delete `/etc/pivot/image-pullspec` during firstboot which happens after OS is upgraded to latest machine-os-content but `machine-config-daemon-firstboot.service` runs early enough and at that time image-pullspec file still exist. Once system reboots after updating OS, `/etc/pivot/image-pullspec` doesn't exist and `/etc/ignition-machine-config-encapsulated.json` exist which leads `machine-config-daemon-firstboot.service`run. That's probably because  systemd `After=ignition-firstboot-complete.service` is not enough to not run `machine-config-daemon-firstboot.service` service if `ignition-firstboot-complete.service` service failed

Reply:
OK, I see, though I think we should probably be deleting the `/etc/ignition-machine-config-encapsulated.json` file too.
But this `BindsTo=` approach is also fine.